component/kafka/group/option.go

```diff
@@ -26,6 +29,52 @@ func FailureStrategy(fs kafka.FailStrategy) OptionFunc {
 	}
 }
 
+// CheckTopic will attempt to:
+//	1. connect to the broker,
+//  2. retrieve the existing topics in the broker
+//  3. if auto-topic creation is disabled, check whether the configured topics exist in the broker
+// If any of the checks above fail the component will exit before starting to consume messages
+func CheckTopic() OptionFunc {
+	return func(c *Component) error {
+		c.topicChecker = func(ctx context.Context, c *Component) error {
```

Comment:
Instead of passing an anonymous function, can't we just extract it as a _real_ function?

Something like:

```go
func CheckTopic() OptionFunc {
	return func(c *Component) error {
		c.topicChecker = checkTopic
		return nil
	}
}

func checkTopic(ctx context.Context, c *Component) error {
```

It feels a bit more readable to me.

Reply:
fair, will do.